### ISSUE TITLE
How can we handle this Exception when running server?

### ISSUE THREAD

**Issue by tientuphudu** [Job79/EasyTcp#49]
hi @Job79 . I'm using 2.04 version. Sometime this exeption make program crash. 
Can you provide some help?

`Application: WhatDogService.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Net.Sockets.SocketException
   at EasyTcp.Server.EasyTcpServer.NotifyOnError(System.Exception)
   at EasyTcp.Server.ServerListener.OnReceiveData(System.IAsyncResult)
   at System.Net.LazyAsyncResult.Complete(IntPtr)
   at System.Net.ContextAwareResult.CompleteCallback(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Net.ContextAwareResult.Complete(IntPtr)
   at System.Net.LazyAsyncResult.ProtectedInvokeCallback(System.Object, IntPtr)
   at System.Net.Sockets.BaseOverlappedAsyncResult.CompletionPortCallback(UInt32, UInt32, System.Threading.NativeOverlapped*)
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32, UInt32, System.Threading.NativeOverlapped*)

`

**Comment by Job79** [comment]
2.04 had some bugs that where fixed by rewriting the whole library (3.0)
This is probably one of them, using the newest version would probably fix this issue. Is there any reason you are still using 2.04? (except different style of documentation)

It probably wouldn't be to hard to rewrite the function that handles the messages in version 2.04, will look into that tomorrow.

**Comment by tientuphudu** [comment]
Yes, new version seam to hard with me on window form. Not clear example and sometime it crash.
So i turn back old project. 
Still in mind newest is better.
I will manager to using new verson you gave us. 
And many thanks for this greate project

**Comment by Job79** [comment]
> Yes, new version seam to hard with me on window form. Not clear example and sometime it crash.
> So i turn back old project.
> Still in mind newest is better.
> I will manager to using new verson you gave us.
> And many thanks for this greate project

Can you give my some points you are struggling with?  Then I will try to include them in the documentation.
I tried to make EasyTcp3 more modular and performant then EasyTcp2, this is probably the reason it is a bit more difficult. 
Or maybe it is because the client and server are disposable?
